extensions/replication/utils/QueueEntry.js

```diff
@@ -1,119 +1,38 @@
+ *
+ * @param {Object} kafkaEntry - entry as read from Kafka queue
+
+ * @return {Object} - an object which inherits from
+ *   {@link QueueEntry} base class
+ */
+function createFromKafkaEntry(kafkaEntry) {
+    try {
+        const record = JSON.parse(kafkaEntry.value);
+        let entry;
+        if (record.bucket === usersBucket) {
+            entry = new BucketQueueEntry(record.key);
+        } else {
+            const mdObj = JSON.parse(record.value);
+            entry = new ObjectQueueEntry(record.bucket, record.key, mdObj);
         }
-        return undefined;
-    }
-
-    static createFromKafkaEntry(kafkaEntry) {
-        try {
-            const record = JSON.parse(kafkaEntry.value);
-            const objMd = JSON.parse(record.value);
-            const entry = new QueueEntry(record.bucket, record.key, objMd);
-            const err = entry.checkSanity();
-            if (err) {
-                return { error: err };
-            }
-            return entry;
-        } catch (err) {
-            return { error: { message: 'malformed JSON in kafka entry',
-                description: err.message } };
+        const err = entry.checkSanity();
+        if (err) {
+            return { error: err };
         }
-    }
-
-    getBucket() {
-        return this.bucket;
-    }
-
-    setBucket(bucket) {
-        this.bucket = bucket;
-        return this;
-    }
-
-    getObjectKey() {
-        return this.objectKey;
-    }
-
-    getObjectVersionedKey() {
-        return this.objectVersionedKey;
-    }
-
-    getUserMetadata() {
-        const metaHeaders = {};
-        const data = this.getValue();
-        Object.keys(data).forEach(key => {
-            if (key.startsWith('x-amz-meta-')) {
-                metaHeaders[key] = data[key];
-            }
-        });
-        if (Object.keys(metaHeaders).length > 0) {
-            return JSON.stringify(metaHeaders);
-        }
-        return undefined;
-    }
-
-    getLogInfo() {
-        return {
-            bucket: this.getBucket(),
-            objectKey: this.getObjectKey(),
-            versionId: this.getVersionId(),
-            isDeleteMarker: this.getIsDeleteMarker(),
-        };
-    }
-
-    toReplicaEntry() {
-        const newEntry = this.clone();
-        newEntry.setBucket(this.getReplicationTargetBucket());
-        newEntry.setReplicationStatus('REPLICA');
-        return newEntry;
-    }
-
-    toCompletedEntry() {
-        const newEntry = this.clone();
-        newEntry.setReplicationStatus('COMPLETED');
-        return newEntry;
-    }
-
-    toFailedEntry() {
-        const newEntry = this.clone();
-        newEntry.setReplicationStatus('FAILED');
-        return newEntry;
+        return entry;
+    } catch (err) {
+        return { error: { message: 'malformed JSON in kafka entry',
+                          description: err.message } };
     }
 }
 
-module.exports = QueueEntry;
+module.exports = {
+    createFromKafkaEntry,
```

Comment:
You should either name this file `createFromKafkaEntry` and export default the method or keep it QueueEntry.